DataFormats/GeometrySurface/interface/ReferenceCounted.h

```diff
@@ -34,28 +32,24 @@ class BasicReferenceCounted
    public:
       BasicReferenceCounted() : referenceCount_(0) {}
       BasicReferenceCounted( const BasicReferenceCounted& /* iRHS */) : referenceCount_(0) {}
+      BasicReferenceCounted(BasicReferenceCounted&&) = default;
+      BasicReferenceCounted& operator=(BasicReferenceCounted&& ) = default;
 
-      const BasicReferenceCounted& operator=( const BasicReferenceCounted& ) {
+      BasicReferenceCounted& operator=( const BasicReferenceCounted& ) {
```

Comment:
Removing const is the proper thing to do since that is really the signature expected by C++. Besides, the member function isn't const anyway.